lite2/client.go

```diff
@@ -89,7 +89,16 @@ func SkippingVerification(trustLevel tmmath.Fraction) Option {
 // current primary is unavailable.
 func Witnesses(providers []provider.Provider) Option {
 	return func(c *Client) {
-		c.witnesses = providers
+		validWitnesses := make([]provider.Provider, 0)
+		for _, witness := range providers {
+			if witness.ChainID() == c.ChainID() {
```

Comment:
nit: maybe a bit nicer if you check if witness.ChainID() != c.ChainID() {c.logger...} then you can drop the else statement.